sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -272,6 +275,32 @@ export class StudentService extends RecordDataModelService<Student> {
       } as StudentAccountApplication;
       await entityManager.getRepository(StudentUser).save(studentUser);
 
+      // Add hold restriction if a partial match was found and the account is not bridge matched
+      if (shouldAddHoldRestriction && !sfasIndividual && !existingStudent) {
+        const holdNote = await this.noteSharedService.createStudentNote(
+          savedStudent.id,
+          NoteType.Restriction,
+          "Restriction added to prevent application completion while potential partial match exists",
+          this.systemUsersService.systemUser.id,
+          entityManager,
+        );
+
+        const holdRestriction = new StudentRestriction();
```

Comment:
Optional. Can we move the block of creating the hold note into a private method.? It will make it easy to read and digest.